Assets/Scripts/Encounters/ShopManager.cs

```diff
@@ -61,6 +67,38 @@ public class ShopManager : GenericSingleton<ShopManager>
         }
     }
 
+    private List<Companion> determineApplicableActiveCompanions(Card cardInfo) {
+        List<Companion> companionList = new();
+
+        //set up companion list
+        foreach (var companion in activeCompanionsVariable.companionList) {
+            //Now this is Epic
```

Comment:
Haha epic indeed.. I think the only alternative here would be to have the card pool lumped into one list on the companionType, maybe a private one that can be retrieved with a getter? I feel like this could be an operation we want to do a lot, might be worth it. Food for thought though, not merge blocking.